src/Services/ConvertToLearnosityService.php

```diff
@@ -255,15 +255,111 @@ private function convertQtiContentPackagesInDirectory($sourceDirectory, $relativ
+                if (isset($convertedContent['rubric'])) {
+                    $rubricReferenceToBeDelete = $this->checkScoringRubricExistInConvertedContent($convertedContent);
+                    if (sizeof($rubricReferenceToBeDelete) > 0) {
+                        foreach ($rubricReferenceToBeDelete as $id => $reference) {
+                            $index = $this->deleteUnusedRubricFromConvertedContent($convertedContent, $reference);
+                            $extraRubricContent = $convertedContent['features'][$index];
+                            unset($convertedContent['features'][$index]);
+                            $scoringRubric = $this->createNewScoringRubricItem($extraRubricContent, $convertedContent['rubric']['reference']);
+                        }
+                    }
+                }
 
+                if (isset($convertedContent['rubric'])) {
+                    unset($convertedContent['rubric']);
+                }
+                $convertedContent = $this->removeUnusedDataFromItem($convertedContent);
                 if (!empty($convertedContent)) {
                     $results['qtiitems'][basename($relativeDir) . '/' . $resourceHref] = $convertedContent;
                 }
+                $scoringRubric = $this->removeUnusedDataFromItem($scoringRubric);
+                if (!empty($scoringRubric)) {
+                    $results['qtiitems'][basename($relativeDir) . '/' . $scoringRubric['item']['reference']] = $scoringRubric;
+                }
             }
         }
         return $results;
     }
 
+    private function removeUnusedDataFromItem($convertedContent)
+    {
+        foreach ($convertedContent['questions'] as $id => $data) {
+            if (isset($data['widget_type'])) {
+                unset($convertedContent['questions'][$id]['widget_type']);
+            }
+            if (isset($data['item_reference'])) {
+                unset($convertedContent['questions'][$id]['item_reference']);
+            }
+            if (isset($data['content'])) {
+                unset($convertedContent['questions'][$id]['content']);
+            }
+            if (isset($convertedContent['item']['metadata']['rubric_reference'])) {
+                unset($convertedContent['item']['metadata']['rubric_reference']);
+            }
+        }
+        return $this->removeUnusedFeatureData($convertedContent);
+    }
+
+    private function removeUnusedFeatureData($convertedContent)
+    {
+        foreach ($convertedContent['features'] as $id => $data) {
+            if (isset($data['widget_type'])) {
+                unset($convertedContent['features'][$id]['widget_type']);
+            }
+            if (isset($data['item_reference'])) {
+                unset($convertedContent['features'][$id]['item_reference']);
+            }
+            if (isset($data['content'])) {
+                unset($convertedContent['features'][$id]['content']);
+            }
+        }
+        return $convertedContent;
+    }
```

Comment:
A warning is sometimes being thrown here:

```
Warning: Illegal string offset 'questions' in /Users/michaelsharman/Documents/workspaces/learnosity-scratch/rakeshqdsa/learnosity-qti/src/Services/ConvertToLearnosityService.php on line 290

Warning: Invalid argument supplied for foreach() in /Users/michaelsharman/Documents/workspaces/learnosity-scratch/rakeshqdsa/learnosity-qti/src/Services/ConvertToLearnosityService.php on line 290

Warning: Illegal string offset 'features' in /Users/michaelsharman/Documents/workspaces/learnosity-scratch/rakeshqdsa/learnosity-qti/src/Services/ConvertToLearnosityService.php on line 311

Warning: Invalid argument supplied for foreach() in /Users/michaelsharman/Documents/workspaces/learnosity-scratch/rakeshqdsa/learnosity-qti/src/Services/ConvertToLearnosityService.php on line 311
```

Suggest something like this to check:

```
private function removeUnusedDataFromItem($convertedContent)
{
    if (is_array($convertedContent) && array_key_exists('questions', $convertedContent)) {
        foreach ($convertedContent['questions'] as $id => $data) {
            if (isset($data['widget_type'])) {
                unset($convertedContent['questions'][$id]['widget_type']);
            }
            if (isset($data['item_reference'])) {
                unset($convertedContent['questions'][$id]['item_reference']);
            }
            if (isset($data['content'])) {
                unset($convertedContent['questions'][$id]['content']);
            }
            if (isset($convertedContent['item']['metadata']['rubric_reference'])) {
                unset($convertedContent['item']['metadata']['rubric_reference']);
            }
        }
    }
    return $this->removeUnusedFeatureData($convertedContent);
}

private function removeUnusedFeatureData($convertedContent)
{
    if (is_array($convertedContent) && array_key_exists('features', $convertedContent)) {
        foreach ($convertedContent['features'] as $id => $data) {
            if (isset($data['widget_type'])) {
                unset($convertedContent['features'][$id]['widget_type']);
            }
            if (isset($data['item_reference'])) {
                unset($convertedContent['features'][$id]['item_reference']);
            }
            if (isset($data['content'])) {
                unset($convertedContent['features'][$id]['content']);
            }
        }
    }
    return $convertedContent;
}
```

Reply:
Done